src/LabelComponent.jsx

```diff
@@ -0,0 +1,158 @@
+  const [filteredSuggestions, setFilteredSuggestions] = useState([]);
+  const [selectedLabels, setSelectedLabels] = useState([null]);
+  const [error, setError] = useState(null);
+
+  useEffect(() => {
+    const fetchData = async () => {
+      try {
+        const response = await fetch(
+          `http://localhost:3000/api/labels?query=${inputValue}`
+        );
+
+        if (!response.ok) {
+          throw new Error("API request failed");
+        }
+
+        const data = await response.json();
+        setLabelsData(data);
+        handleInputChange(inputValue);
+      } catch (error) {
+        setError(error.message);
+      }
+    };
+
+    fetchData();
+  }, [inputValue]);
+
+  const handleInputChange = (input) => {
+    if (!labelsData || !labelsData.member) {
+      return;
+    }
+    const filtered = input.trim()
+      ? labelsData.member
+          .filter(
+            (label) =>
+              label.name.toLowerCase().includes(input.toLowerCase()) &&
+              label.visibility === "visible"
+          )
+          .map((label) => label.name)
+      : [];
+
+    setFilteredSuggestions(filtered);
+  };
+
+  const handleAddLabel = async (label) => {
+    if (selectedLabels.includes(label) || !label.trim()) return;
+
+    try {
+      const labelName = encodeURIComponent(label);
+      const response = await fetch(
+        `http://localhost:3000/api/events/:eventId/labels/${labelName}`,
+        {
+          method: "PUT",
+        }
+      );
+
+      if (response.status === 204) {
+        const eventResponse = await fetch(
+          `http://localhost:3000/api/events/:eventId`
+        );
+        if (eventResponse.ok) {
+          const eventData = await eventResponse.json();
+          setSelectedLabels(eventData.labels); // Update selectedLabels based on fetched data
+        } else {
+          throw new Error("Failed to fetch updated event details");
+        }
+
+        // Clear the input and suggestions
+        setInputValue("");
+        setFilteredSuggestions([]);
+      } else {
+        throw new Error("Failed to add label");
+      }
+    } catch (error) {
+      setError(error.message);
+    }
+  };
+
+  const handleRemoveLabel = async (label) => {
+    try {
+      const labelName = encodeURIComponent(label);
+      const response = await fetch(
+        `http://localhost:3000/api/events/:eventId/labels/${labelName}`,
+        {
+          method: "DELETE",
+        }
+      );
+
+      if (!response.ok) {
+        throw new Error("Failed to remove label");
+      }
+
+      // Fetch the updated event details to get the remaining labels
+      const eventResponse = await fetch(
+        `http://localhost:3000/api/events/:eventId`
+      );
+      if (eventResponse.ok) {
+        const eventData = await eventResponse.json();
+        setSelectedLabels(eventData.labels);
+      } else {
+        throw new Error("Failed to fetch updated event details");
+      }
```

Comment:
Refetching the event is done multiple times. This can be abstracted into its own function.